core/qthelper.cpp

```diff
@@ -1642,6 +1642,15 @@ char *copy_qstring(const QString &s)
 	return strdup(qPrintable(s));
 }
 
+// function to call to allow the UI to show updates for longer running activities
+void (*uiNotificationCallback)(QString msg) = nullptr;
+
+void uiNotification(const QString &msg)
+{
+	if (uiNotificationCallback != nullptr)
+		uiNotificationCallback(QString(msg));
```

Comment:
Casting QString to QString here. Seems redundant.

Reply:
that's because initially I had considered doing what you asked for an have a single notification mechanism, which required it being able to be called from C code, so I passed in a `const char *` that was then cast to `QString`.
And then that whole thing pissed me off because we would use `tr()` to create a localized string (since this is user visible), then extract the `char *` from that, only to then pass it around again as `QString`.

Reply:
fixed